docs/docs/user-guide/background/megatron_datasets.md

```diff
@@ -56,6 +56,42 @@ for sample in MultiEpochDatasetResampler(dataset, num_epochs=3, shuffle=True):
     [PRNGResampleDataset][bionemo.core.data.resamples.PRNGResampleDataset] offers a simple solution for shuffling a
     dataset with replacement in O(1) memory.
 
+## Training Resumption
+To ensure identical behavior with and without job interruption, BioNeMo provides [MegatronDataModule][bionemo.llm.data.datamodule.MegatronDataModule] to save and load state dict for training resumption, and provides [WrappedDataLoader][nemo.lightning.data.WrappedDataLoader] to add a `mode` attribute to [DataLoader][torch.utils.data.DataLoader].
+
+```python
+class MyDataModule(MegatronDataModule):
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        ...
+
+    def train_dataloader(self):
```

Comment:
TODO add `self.update_init_global_step()` just before the WrappedDataloader return.